project_euler/problem_80/sol1.py

```diff
@@ -0,0 +1,57 @@
+"""
+https://projecteuler.net/problem=80
+
+It is well known that if the square root of a natural number is not an integer,
+then it is irrational. The decimal expansion of such square roots is infinite
+without any repeating pattern at all.
+
+The square root of two is 1.41421356237309504880...,
+and the digital sum of the first one hundred decimal digits is 475.
+
+For the first one hundred natural numbers, find the total of the digital
+sums of the first one hundred decimal digits for all the irrational square roots.
+"""
+
+
+from decimal import Decimal, getcontext
+
+
+def sqrt_with_accuracy(num: int, accuracy: int) -> str:
+    """
+    Function to take the square root of a number with given accuracy.
+    Returns string of answer without a dot.
+    >>> sqrt_with_accuracy(2, 30)
+    '141421356237309504880168872420'
+    >>> sqrt_with_accuracy(80, 100)
+    '8944271909999158785636694674925104941762473438446102897083588981642083702551219597657657633515129099'
+    """
+    getcontext().prec = accuracy + 5
+    ans = str(Decimal(num).sqrt())
+    ans = ans.replace(".", "")
+    return ans[:accuracy]
+
+
+def solution():
```

Comment:
Please break this into two functions.  The first should return the value for just one number and should have a doctest that proves the answer on lines 8 and 9.  The second should use the first to calculate the numbers between 2 and 100.

Is there a source that tells us that 40886 is the correct answer?

Reply:
I've split the functions, but I'm unsure where to find a source other than entering it into https://projecteuler.net/problem=80 and showing that 40886 is indeed the answer.

Reply:
Why get rid of `sqrt_with_accuracy()`?  That is a cool function and is useful in other contexts.  Let’s bring that one back and we can feed it’s output into a much simpler `sum_of_digits()’.